main.go

```diff
@@ -235,6 +235,9 @@ func main() {
 	log.Infof("Extracting cache archive")
 
 	if err := extractCacheArchive(cacheRecorderReader); err != nil {
+		log.Debugf("Failed to uncompress cache archive stream: %s", err)
```

Comment:
This could be log.Errorf as is an error even if there is a fallback logic. I am also not sure if we will know if there are failures as the fallback just returns after printing out the error.